packager/packager.cc

```diff
@@ -32,10 +33,16 @@ bool isTestNamespace(const core::NameRef ns) {
     return ns == core::packages::PackageDB::TEST_NAMESPACE;
 }
 
-bool visibilityApplies(const core::packages::VisibleTo &vt, absl::Span<const core::NameRef> name) {
+bool visibilityApplies(const core::GlobalState &gs, const core::packages::VisibleTo &vt,
+                       core::packages::MangledName name) {
     if (vt.visibleToType == core::packages::VisibleToType::Wildcard) {
         // a wildcard will match if it's a proper prefix of the package name
-        return vt.packageName == name.subspan(0, vt.packageName.size());
+        // TODO(jez) Replace this with looking up in ownership nesting hierarchy
+        auto curPkgName = name.mangledName.shortName(gs);
+        auto prefix = vt.packageName.mangledName.shortName(gs);
+        return absl::StartsWith(curPkgName, prefix) &&
+               // Have to check '_' to make sure we're on a package boundary.
+               (curPkgName.size() == prefix.size() || curPkgName[prefix.size()] == '_');
```

Comment:
I assume this to prevent the case where the prefix is `Foo::Bar::*` and the name is `Foo::B`, since `Foo_Bar` starts with `Foo_B`, but `Foo_Bar[5] != _`?

If there isn't already, we should add a test case for this.

Reply:
Good catch, I've added a test.